src/Grpc.NetCore.HttpClient/Internal/GrpcCall.cs

```diff
@@ -415,7 +415,11 @@ private void DeadlineExceeded(object state)
 
         private static Status GetStatusCore(HttpResponseMessage httpResponseMessage)
         {
-            string grpcStatus = GetHeaderValue(httpResponseMessage.TrailingHeaders, GrpcProtocolConstants.StatusTrailer);
+            // A gRPC server may return trailers in the headers when the response stream is empty
+            // For example, C Core server returns them together in the empty_stream interop test
+            var grpcStatus = GetHeaderValue(httpResponseMessage.TrailingHeaders, GrpcProtocolConstants.StatusTrailer)
+                ?? GetHeaderValue(httpResponseMessage.Headers, GrpcProtocolConstants.StatusTrailer);
```

Comment:
This is correct, but too permissive.
Looks like this can potentially allow a malformed RPC where  grpc-status is in the headers, but there is also payload and there are some trailers. In the "Trailers-Only case" from the spec, the `TrailingHeaders` need to be empty and there must be no payload. Let's add some check in that sense.

In c-core, it seems that we only allow "Trailers-only" case if  the headers frame being analyzed is the last frame of the stream.
https://github.com/grpc/grpc/blob/50e16f6d3a99335f09060d62cfa495c8977e3b2b/src/core/ext/transport/chttp2/transport/parsing.cc#L630
https://github.com/grpc/grpc/blob/50e16f6d3a99335f09060d62cfa495c8977e3b2b/src/core/ext/transport/chttp2/transport/parsing.cc#L553

Reply:
Help me please I’ve gotten assistance from github before and now I have several questions but am very new to coding as I used to do it long time ago. Would appreciate input on where and how to start so I can participate and get my questions answered